src/Symfony/Component/Pgp/PgpEncrypter.php

```diff
@@ -0,0 +1,266 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Pgp;
+
+use Symfony\Component\Mime\Header\Headers;
+use Symfony\Component\Mime\Header\MailboxListHeader;
+use Symfony\Component\Mime\Message;
+use Symfony\Component\Mime\Part\Multipart\MixedPart;
+use Symfony\Component\Pgp\Exception\BadPassphraseException;
+use Symfony\Component\Pgp\Exception\FileException;
+use Symfony\Component\Pgp\Exception\GeneralException;
+use Symfony\Component\Pgp\Exception\KeyNotFoundException;
+use Symfony\Component\Pgp\Mime\Part\Multipart\PgpEncryptedPart;
+use Symfony\Component\Pgp\Mime\Part\Multipart\PgpSignedPart;
+use Symfony\Component\Pgp\Mime\Part\PgpEncryptedInitializationPart;
+use Symfony\Component\Pgp\Mime\Part\PgpEncryptedMessagePart;
+use Symfony\Component\Pgp\Mime\Part\PgpKeyPart;
+use Symfony\Component\Pgp\Mime\Part\PgpSignaturePart;
+use Symfony\Component\Pgp\Mime\Traits\PgpSigningTrait;
+
+/*
+ * @author PuLLi <the@pulli.dev>
+ */
+final class PgpEncrypter
+{
+    use PgpSigningTrait;
+
+    private \Crypt_GPG $gpg;
+
+    private Headers $headers;
+
+    private ?string $signingKey = null;
+
+    public string $signature = '';
+
+    public string $signed = '';
+
+    /**
+     * @throws FileException
+     * @throws GeneralException
+     */
+    public function __construct(array $options = [])
+    {
+        try {
+            $this->gpg = new \Crypt_GPG(
+                array_merge(
+                    $options,
+                    [
+                        'cipher-algo' => 'AES256',
+                        'digest-algo' => 'SHA512',
+                    ],
+                )
+            );
+        } catch (\Crypt_GPG_FileException $e) {
+            throw new FileException($e->getMessage());
```

Comment:
Please attach the original exception as previous exception. It might make things easier to debug.

Reply:
This applies to other places doing exception wrapping.